elastic/eui
# [Theme] Base theme setup

## Summary

closes [#124](https://github.com/elastic/eui-private/issues/124)

>[!IMPORTANT]
>This PR is currently still in `Draft` mode, as it needs alignment on some package/dependency specific setups/architectural choices.

📖 See additional information [here](https://github.com/elastic/eui-private/issues/124#issuecomment-2368629284).

## QA

- [x] checkout this PR and verify packages build and run as expected:
  - `/eui`: run `build:workspaces` to build workspace dependencies then check that our common development tools/scripts work as expected
    - Storybook: `yarn storybook` and `yarn build-storybook`
    - EUI docs: `yarn start` and `yarn build-docs`
  - `/website`: run `build:workspaces` and then test that EUI+ docs works as expected via `yarn start` and `yarn build`

### Deployments & theme switcher testing

By default the theme switcher is only enabled for `development` environments.

Storybook: https://eui.elastic.co/pr_8030/storybook/?path=/story/navigation-euibutton--playground
EUI docs: https://eui.elastic.co/pr_8030/
EUI+: https://eui.elastic.co/pr_8030/new-docs/
